index.ts

```diff
@@ -229,7 +214,10 @@ class DatabaseFake {
     const _creationTime =
       now <= this._lastCreationTime ? this._lastCreationTime + 0.001 : now;
     this._lastCreationTime = _creationTime;
-    this._writes[_id] = {
+    if (!this._writes.length) {
```

Comment:
```suggestion
    if (this._writes.length === 0) {
```

Reply:
Maybe make a private method that's like `_addWrite(id, update)` (better name pls)